controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -120,134 +160,384 @@ func (r *StorageConsumerReconciler) Reconcile(ctx context.Context, request recon
+}
 
-	if r.storageConsumer.GetDeletionTimestamp().IsZero() {
+func (r *StorageConsumerReconciler) reconcileCephRadosNamespace() error {
+	blockPools := &rookCephv1.CephBlockPoolList{}
+	if err := r.List(r.ctx, blockPools, client.InNamespace(r.namespace)); err != nil {
+		return err
+	}
 
-		// A provider cluster already has a NooBaa system and does not require a NooBaa account
-		// to connect to a remote cluster, unlike client clusters.
-		// A NooBaa account only needs to be created if the storage consumer is for a client cluster.
-		clusterID := util.GetClusterID(r.ctx, r.Client, &r.Log)
-		if clusterID != "" && !strings.Contains(r.storageConsumer.Name, clusterID) {
-			if err := r.reconcileNoobaaAccount(); err != nil {
-				return reconcile.Result{}, err
+	// ensure for this consumer a rados namespace is created in every blockpool
+	for idx := range blockPools.Items {
+		bp := &blockPools.Items[idx]
+		if bp.Name == "builtin-mgr" {
+			continue
+		}
+
+		rns := &rookCephv1.CephBlockPoolRadosNamespace{}
+		rns.Name = func() string {
+			if r.isUpdatedProvider && strings.HasPrefix(bp.Name, r.storageCluster.Name) {
+				return r.radosnamespaceName
+			}
+			return fmt.Sprintf("%s-%s", bp.Name, r.radosnamespaceName)
+		}()
+
+		rns.Namespace = r.namespace
+
+		if _, err := ctrl.CreateOrUpdate(r.ctx, r.Client, rns, func() error {
+			// if rns was created in provider mode storagerequest will be the controller owner
+			// as only single controller owner is permitted remove the old and place consumer as the owner
+			if existing := metav1.GetControllerOfNoCopy(rns); existing != nil &&
+				existing.Kind != r.storageConsumer.Kind {
+				existing.BlockOwnerDeletion = nil
+				existing.Controller = nil
 			}
+			if err := r.own(rns); err != nil {
+				return err
+			}
+			rns.Spec.Name = r.radosnamespaceName
+			return nil
+		}); err != nil {
+			return err
 		}
 
-		cephResourcesReady := true
-		for _, cephResource := range r.storageConsumer.Status.CephResources {
-			if cephResource.Phase != "Ready" {
-				cephResourcesReady = false
-				break
+		phase := ""
+		if rns.Status != nil {
+			phase = string(rns.Status.Phase)
+		}
+		r.setCephResourceStatus(rns.Name, "CephBlockPoolRadosNamespace", phase)
+
+	}
+
+	//TODO: set this properly
+	osdCaps := fmt.Sprintf("profile rbd namespace=%s", r.radosnamespaceName)
+	if r.isLocalConsumer {
+		osdCaps = "profile rbd namespace"
+	}
+
+	cephClientRbdProvisioner := &rookCephv1.CephClient{}
+	cephClientRbdProvisioner.Name = generateClientName("rbd", "provisioner", string(r.storageConsumer.UID))
+	cephClientRbdProvisioner.Namespace = r.namespace
+	if _, err := ctrl.CreateOrUpdate(r.ctx, r.Client, cephClientRbdProvisioner, func() error {
+		if err := r.own(cephClientRbdProvisioner); err != nil {
+			return err
+		}
+		util.AddAnnotation(
+			cephClientRbdProvisioner,
+			CephClientSecretNameAnnotationKey,
+			generateClientName("rbd", "provisioner", string(r.storageConsumer.UID)),
+		)
```

Comment:
we don't need this now as cephclient per each ceph user (provisioner/node) per each consumer will not collide even if all of them are created in same namespace except for local client, see if the secret name can be customized in cephclient CR.